java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java

```diff
@@ -1134,7 +1183,10 @@ public void setRowIndexCols(boolean[] rowIndexCols) {
             if (entry == null) {
               throw new AssertionError("RG is not populated for " + columnIx + " rg " + rowGroup);
             }
-            OrcProto.ColumnStatistics stats = entry.getStatistics();
+            OrcProto.ColumnStatistics stats = EMPTY_COLUMN_STATISTICS;
+            if (entry.hasStatistics()) {
+              stats = entry.getStatistics();
```

Comment:
Here is the key to fix.

When no column statistics are provided, entry.getStatistics() does not return null, but a default object that we must first determine by entry.hasStatistics() before getting it.

JsonFileDump has the same misuse as FileDump, I'll fix them later